lib/eventsimple/event_dispatcher.rb

```diff
@@ -77,16 +78,18 @@ class ReactorSet
       attr_reader :sync, :async
 
       def initialize
-        @sync = Set.new
-        @async = Set.new
+        @sync = []
+        @async = []
       end
 
       def add_sync(reactors)
-        @sync += reactors
+        new_reactors = reactors - @sync
+        @sync += new_reactors
       end
 
       def add_async(reactors)
-        @async += reactors
+        new_reactors = reactors - @async
+        @async += new_reactors
```

Comment:
I think this is equivalent to using the [| operator](https://ruby-doc.org/core-2.5.0/Array.html#method-i-7C).

```suggestion
        @async |= reactors
```

Reply:
oh nice! I didn't know this operator existed. TY